### PR TITLE
seventeenlands: init at 0.1.42

### PR DESCRIPTION
**Pull request by sephii**
## Description of changes

Client to visualize data from Magic: The Gathering Arena matches.

Home page: https://www.17lands.com/
Source code: https://github.com/rconroy293/mtga-log-client

I used the PyPI release instead of GitHub because GitHub releases are not tagged.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/se/seventeenlands/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  doCheck = false;
+
+  propagatedBuildInputs = with python3.pkgs; [
+    python-dateutil
+    requests
+    tkinter
+  ];
+
+  meta = with lib; {
+    description = "Simple client for passing relevant events from MTG Arena logs to the 17Lands REST endpoint.";
```
Comment (Aleksanaa) [inline pkgs/by-name/se/seventeenlands/package.nix:25]:
```suggestion
    description = "Passing relevant events from MTG Arena logs to the 17Lands REST endpoint, also named mtga-log-client";
```

File: pkgs/by-name/se/seventeenlands/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "seventeenlands";
+  version = "0.1.42";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-P/imV4vvyd6wgjqXzgfIAURFtFhLwX1eS8eiPl79oZk=";
+  };
+
+  # No tests
+  doCheck = false;
```
Comment (Aleksanaa) [inline pkgs/by-name/se/seventeenlands/package.nix:16]: Then at least add a `pythonImportsCheck = [ "seventeenlands" ];` (test if it works)

File: pkgs/by-name/se/seventeenlands/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  propagatedBuildInputs = with python3.pkgs; [
+    python-dateutil
+    requests
+    tkinter
+  ];
+
+  meta = with lib; {
+    description = "Simple client for passing relevant events from MTG Arena logs to the 17Lands REST endpoint.";
+    homepage = "https://www.17lands.com/";
+    license = licenses.gpl3;
```
Comment (Aleksanaa) [inline pkgs/by-name/se/seventeenlands/package.nix:27]:
```suggestion
    license = licenses.gpl3Only;
```

https://github.com/rconroy293/mtga-log-client?tab=readme-ov-file#notes

File: pkgs/by-name/se/seventeenlands/package.nix (hunk 1 above)
Comment (sephii) [reply pkgs/by-name/se/seventeenlands/package.nix]: Thanks for the suggestion. I’ve updated the description but I left the "client" part, which I think is important.

**Pull request merged**